### ISSUE TITLE
Some simple commands fail (quoting issue?)

### ISSUE THREAD

**Issue by ndemou** [xonsh/xonsh#2960]
This simple command should print test...
```
echo test |sed -e 's/.<-/\n<-/g'
```
...but instead I get this error:
```
/bin/sed: -e expression #1, char 6: unterminated `s' command
```

I'm using xonsh since many months now and from time to time I notice that when I type bash commands with a bit complex quoting xonsh will fail. I noticed it again today at xonsh v0.7 and decided to report it so I upgraded and verified that it's still buggy on v0.8.4 (unfortunately the upgrade doubled my startup time but that's another [issue](https://github.com/xonsh/xonsh/issues/2394#issuecomment-447364888)).

**Comment by scopatz** [comment]
This is because of the way that Python strings work.  Using a raw string as the substation command worked for me:

```console
$ echo test |sed -e 's/.<-/\n<-/g'
/bin/sed: -e expression #1, char 6: unterminated `s' command
$ echo test |sed -e r's/.<-/\n<-/g'
test
```

Basically, the `\n` is being interpreted as a true new line in the original example, which isn't proper sed syntax.

**Comment by ndemou** [comment]
Oh! boy so obvious. Sorry for the stupid report. 

From time to time bash is the right tool for the job and I have two options:
1. stay on xonsh but enclose bash commands like this `bash -c r'''... '''`
2. switch to bash for a while

Typing `bash -c r''' '''` is boring and I end up running bash inside xonsh. Which I regret when I'm trying to find that useful one-liner from 2 weeks ago since my command history is split between xonsh and bash.

Is there a better way?

**Comment by scopatz** [comment]
@ndemou - are there specific cases where you think bash is the right tool?  I'd love to hear where you think xonsh is falling short and try to address them! 

In this case, I think xonsh is doing the right thing still, by being consistent and predictable with strings. We have had a lot of debate about this in the past, but ended up with the discussion here: https://xon.sh/tutorial_subproc_strings.html

**Comment by ndemou** [comment]
Sorry @scopatcz I somehow overlooked your question. So regarding "are there specific cases where you think bash [instead of xonsh] is the right tool?":

**TLDR**: Not any more, you just need self-restraint and some time to allow xonsh to become rock solid.

**Explanation**: During the last 3 months I only had to leave xonsh due to bugs. Luckily xonsh is mature enough that I've hit just 2 bugs in this period :-) Unluckily xonsh is mature enough that the bugs that remain are those "I-want-to-pull-my-hairs-out" cases where it isn't even obvious that xonsh is the culprit and you spent a lot of time figuring it out :-( At such times one questions his decision to use on a daily basis a tool that is not rock solid. Please don't get offended! It's not your fault that xonsh is not rock solid. This needs time. The only reason I'm sharing these feelings is because: 

- I *guess* that these bugs also hit enough other users 
- I *guess* that they make them feel more ore less the same 
- once xonsh becomes rock solid I want you to remember such stories so that you can be more conservative and self-restrained when it comes to making serious changes to the code-base :-)

BTW issue #3044 was one of those 2 bugs that hit me and last month there was another bug where some TUI program was not working and I realized it was due to xonsh (!!!?) I'm now sorry that I didn't fill a bug right there and then (because I remember nothing at all now) but sometimes my relation with xonsh is like a relation with a friend and my reactions are all but rational: I'm hard pressed to catch a deadline, use my friend xonsh to help, he let's me down and I get mad at him and stop talking to him for a few days (i.e. I don't fill a bug report to help him even though I know this will be for my own good) */me leaves to go take his daily psychiatric medication*.


On a side note: I believe I can write a "How to comfortably switch from bash to xonsh" page for the documentation to help bash users by collecting all the important info in on one page. Do you think it would be useful?
